src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -1092,6 +1092,31 @@ object RichContextRDDRegionValue {
 }
 
 class RichContextRDDRegionValue(val crdd: ContextRDD[RVDContext, RegionValue]) extends AnyVal {
+  def boundary: ContextRDD[RVDContext, RegionValue] =
+    crdd.cmapPartitionsAndContext { (consumerCtx, part) =>
+      val producerCtx = consumerCtx.freshContext
+      val it = part.flatMap(_ (producerCtx))
+      new Iterator[RegionValue]() {
+        private[this] var cleared: Boolean = false
+
+        def hasNext = {
+          if (!cleared) {
+            cleared = true
+            producerCtx.region.clear()
```

Comment:
I just copied the implementation of `boundary` from `RVD.crddBoundary`, because it only depended on the `RVD`'s `crdd`, and I needed to use it somewhere I only had a `ContextRDD`. But my understanding is that the `it.hasNext` call on the line below will construct `it`'s next value (or at least try to), so the data in the region currently will leak if we don't clear it now.